client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/index.tsx

```diff
@@ -44,8 +47,23 @@ const Launchpad: Step = ( { navigation, flow }: LaunchpadProps ) => {
 		[]
 	);
 
-	if ( ! isLoggedIn ) {
+	const redirectToSiteHome = useCallback( ( siteSlug: string | null, flow: string | null ) => {
+		recordTracksEvent( 'calypso_launchpad_redirect_to_home', { flow: flow } );
 		window.location.replace( `/home/${ siteSlug }` );
+	}, [] );
+
+	const areLaunchpadTasksCompleted = useCallback(
+		( flow: string | null ) => {
+			if ( isNewsletterFlow( flow ) ) {
+				return checklist_statuses?.first_post_published;
+			}
+			return isSiteLaunched || checklist_statuses?.site_launched;
+		},
+		[ checklist_statuses?.first_post_published, checklist_statuses?.site_launched, isSiteLaunched ]
+	);
+
```

Comment:
The addition and use of areLaunchpadTasksCompleted() is the main change in this PR. For newsletter sites, if first_post_published is done (the last launchpad task for that flow), then Launchpad should already be complete/already. Likewise, for all other flows, if site_launched is done, or the site is actually already launched, then launchpad should be complete/off already.

So we check those here, and below if areLaunchpadTasksCompleted(), we update launchpad_screen to 'off' and redirect to home.